### PR TITLE
Introduce permission system restrictions

### LINKED ISSUE #62: Permission system

### PR DESCRIPTION
**Pull request by Brutus5000**
(fixes #62)

### CODE REVIEW

File: src/main/java/com/faforever/moderatorclient/api/dto/UserGroup.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+@Getter
+@Setter
+@Type(UserGroup.TYPE_NAME)
+public class UserGroup extends AbstractEntity {
+
+    public static final String TYPE_NAME = "userGroup";
+
+    private String technicalName;
+    private String nameKey;
+    private boolean public_;
```
Comment (axel1200) [inline src/main/java/com/faforever/moderatorclient/api/dto/UserGroup.java]: with _?

Comment (Brutus5000) [reply src/main/java/com/faforever/moderatorclient/api/dto/UserGroup.java]: `public` is a reserved keyword in Java, but not in MySQL. Do you have a better idea?

**Comment by bukajsytlos** [comment]
published?

**Comment by Brutus5000** [comment]
That is semantically not the same.

**Pull request merged**